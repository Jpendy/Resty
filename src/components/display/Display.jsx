import React from 'react';
import PropTypes from 'prop-types';

const Display = ({ arr }) => {

  let listElement = arr.map(item => {
    return (
      <li key={item}>
        {item}
      </li>
    );
  });

  return (
    <ul>
      {listElement}
    </ul>
  );
};


Display.propTypes = {
  arr: PropTypes.arrayOf(PropTypes.object.isRequired)
};

export default Display;
