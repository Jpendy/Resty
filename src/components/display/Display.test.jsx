import React from 'react';
import { shallow } from 'enzyme';
import Display from './Display';

describe('Display component', () => {
  it('renders Display', () => {
    const wrapper = shallow(<Display arr={[{ thing1: 'thing' }, { thing: 'thing' }]}  />);
    expect(wrapper).toMatchSnapshot();
  });
});
