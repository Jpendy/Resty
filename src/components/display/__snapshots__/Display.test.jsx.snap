// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component renders Display 1`] = `
<ul>
  <li
    key="[object Object]"
  >
    <Component />
  </li>
  <li
    key="[object Object]"
  >
    <Component />
  </li>
</ul>
`;
